app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,82 @@
+class RecipesController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_recipe, except: %i[index new create public_list]
+
+  # GET /recipes or /recipes.json
+  def index
+    @recipes = Recipe.where(user_id: current_user.id)
+  end
+
+  def public_list
+    # @recipes = Recipe.where(public: true).includes(:user)
+    @recipes = Recipe.where(public: true)
+  end
+
+  # GET /recipes/1 or /recipes/1.json
+  def show
+    @modify = current_user.id == @recipe.user_id
+  end
+
+  # GET /recipes/new
+  def new
+    @recipe = Recipe.new
+  end
+
+  # GET /recipes/1/edit
+  def edit; end
+
+  # POST /recipes or /recipes.json
+  def create
+    @recipe = Recipe.new(recipe_params)
+
+    respond_to do |format|
+      if @recipe.save
+        format.html { redirect_to recipe_url(@recipe), notice: 'Recipe was successfully created.' }
+        format.json { render :show, status: :created, location: @recipe }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @recipe.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /recipes/1 or /recipes/1.json
+  def update
+    respond_to do |format|
+      if @recipe.update(recipe_params)
+        format.html { redirect_to recipe_url(@recipe), notice: 'Recipe was successfully updated.' }
+        format.json { render :show, status: :ok, location: @recipe }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @recipe.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def update_public
+    @recipe.update(public: !@recipe.public)
+    redirect_to recipe_url(@recipe)
+  end
+
+  # DELETE /recipes/1 or /recipes/1.json
+  def destroy
+    @recipe.destroy
+
+    respond_to do |format|
+      format.html { redirect_to recipes_url, notice: 'Recipe was successfully destroyed.' }
+      format.json { head :no_content }
+    end
+  end
```

Comment:
- [ ] When I add a recipe to my shopping list and try to delete it, I get an error, kindly fix that
![Screenshot from 2023-09-15 22-08-34](https://github.com/gdumani/recipify/assets/66097371/7421b45d-d98b-4cb7-a21b-f4410ee947c3)

Reply:
Done! Thank you